src/main/java/org/openrewrite/java/spring/boot3/RemoveConstructorBindingAnnotation.java

```diff
@@ -43,12 +45,21 @@ public String getDescription() {
     @Override
     public JavaIsoVisitor<ExecutionContext> getVisitor() {
```

Comment:
Please feel free to adjust the code as you'd like.

The main changes are:

- Use message instead of local variables.
- Replaced visitAnnotation with `maybeAddComment`.
- Prevent unnecessary cycles caused by `annotation1.withComments(list)`.
- Added structured Javadoc to allow `NormalizeLineBreaksVisitor` to autoformat when necessary.

```java
    @Override
    public JavaIsoVisitor<ExecutionContext> getVisitor() {
        return new JavaIsoVisitor<ExecutionContext>() {

            @Override
            public J.ClassDeclaration visitClassDeclaration(J.ClassDeclaration classDecl, ExecutionContext executionContext) {
                List<J.MethodDeclaration> constructors = classDecl.getBody().getStatements().stream()
                        .filter(o -> o instanceof J.MethodDeclaration)
                        .map(o -> (J.MethodDeclaration) o)
                        .filter(J.MethodDeclaration::isConstructor)
                        .collect(Collectors.toList());

                if (constructors.size() == 1 &&
                        constructors.get(0).getLeadingAnnotations().stream()
                                .anyMatch(a -> TypeUtils.isOfClassType(a.getType(), ANNOTATION_CONSTRUCTOR_BINDING))) {
                    executionContext.putMessage("CONSTRUCTOR_TO_UPDATE", constructors.get(0));
                }

                J.ClassDeclaration c = super.visitClassDeclaration(classDecl, executionContext);
                if (c.getLeadingAnnotations().stream().anyMatch(a -> TypeUtils.isOfClassType(a.getType(), ANNOTATION_CONFIG_PROPERTIES))) {
                    c = c.withLeadingAnnotations(ListUtils.map(c.getLeadingAnnotations(), anno -> {
                        if (TypeUtils.isOfClassType(anno.getType(), ANNOTATION_CONSTRUCTOR_BINDING)) {
                            if (constructors.size() <= 1) {
                                maybeRemoveImport(ANNOTATION_CONSTRUCTOR_BINDING);
                                return null;
                            }

                            return anno.withComments(maybeAddComment(anno.getComments()));
                        }
                        return anno;
                    }));
                }
                return c;
            }

            private List<Comment> maybeAddComment(List<Comment> comments) {
                String message = "You need to remove ConstructorBinding on class level and move it to appropriate";
                if (comments.isEmpty() || comments.stream()
                        .filter(o -> o instanceof Javadoc.DocComment)
                        .map(o -> (Javadoc.DocComment) o)
                        .flatMap(o -> o.getBody().stream().filter(j -> j instanceof Javadoc.Text))
                        .noneMatch(o -> o.print(getCursor()).equals(message))) {

                    List<Javadoc> javadoc = new ArrayList<>();
                    javadoc.add(new Javadoc.LineBreak(randomId(), "\n * ", Markers.EMPTY));
                    javadoc.add(new Javadoc.Text(UUID.randomUUID(), Markers.EMPTY, "TODO:"));
                    javadoc.add(new Javadoc.LineBreak(randomId(), "\n * ", Markers.EMPTY));
                    javadoc.add(new Javadoc.Text(UUID.randomUUID(), Markers.EMPTY, message));
                    javadoc.add(new Javadoc.LineBreak(randomId(), "\n * ", Markers.EMPTY));
                    javadoc.add(new Javadoc.Text(UUID.randomUUID(), Markers.EMPTY, "constructor."));
                    javadoc.add(new Javadoc.LineBreak(randomId(), "\n ", Markers.EMPTY));

                    List<Comment> newComments = new ArrayList<>(comments);
                    newComments.add(new Javadoc.DocComment(UUID.randomUUID(), Markers.EMPTY, javadoc, "\n"));
                    return newComments;
                }
                return comments;
            }

            @Override
            public J.MethodDeclaration visitMethodDeclaration(J.MethodDeclaration method, ExecutionContext executionContext) {
                J.MethodDeclaration methodDeclaration = super.visitMethodDeclaration(method, executionContext);

                J.MethodDeclaration constructorToUpdate = executionContext.getMessage("CONSTRUCTOR_TO_UPDATE");
                if (method == constructorToUpdate) {
                    methodDeclaration = methodDeclaration.withLeadingAnnotations(ListUtils.map(method.getLeadingAnnotations(), anno -> {
                        if (TypeUtils.isOfClassType(anno.getType(), ANNOTATION_CONSTRUCTOR_BINDING)) {
                            maybeRemoveImport(ANNOTATION_CONSTRUCTOR_BINDING);
                            return null;
                        }
                        return anno;
                    }));
                    executionContext.pollMessage("CONSTRUCTOR_TO_UPDATE");
                }
                return methodDeclaration;
            }
        };
    }
```